xdsl/dialects/riscv.py

```diff
@@ -0,0 +1,765 @@
+
+    @staticmethod
+    @builder
+    def from_register(register: Register) -> RegisterAttr:
+        return RegisterAttr(register)
+
+
+@irdl_attr_definition
+class LabelAttr(Data[str]):
+    name = "riscv.label"
+
+    @staticmethod
+    def parse_parameter(parser: Parser) -> str:
+        data = parser.parse_while(lambda x: x != ">")
+        return data
+
+    @staticmethod
+    def print_parameter(data: str, printer: Printer) -> None:
+        printer.print_string(data)
+
+    @staticmethod
+    @builder
+    def from_str(name: str) -> LabelAttr:
+        return LabelAttr(name)
+
+
+Op = TypeVar("Op", bound=Operation)
+
+
+class Riscv1Rd1Rs1ImmOperation(Operation):
+    rd: OpAttr[RegisterAttr]
+    rs1: OpAttr[RegisterAttr]
+    immediate: OpAttr[IntegerAttr]
+
+    @classmethod
+    def get(cls: Type[Op], rd, rs1, immediate) -> Op:
+        if isinstance(immediate, int):
+            immediate = IntegerAttr.from_int_and_width(immediate, 64)
+        return cls.build(attributes={
+            "rd": rd,
+            "rs1": rs1,
+            "immediate": immediate
+        })
+
+
+class Riscv1Rd1Rs1OffOperation(Operation):
+    rd: OpAttr[RegisterAttr]
+    rs1: OpAttr[RegisterAttr]
+    offset: OpAttr[AnyOf([IntegerAttr, LabelAttr])]
+
+    @classmethod
+    def get(cls: Type[Op], rd, rs1, offset) -> Op:
+        if isinstance(offset, int):
+            offset = IntegerAttr.from_int_and_width(offset, 64)
+        if isinstance(offset, str):
+            offset = LabelAttr.from_str(offset)
+        return cls.build(attributes={"rd": rd, "rs1": rs1, "offset": offset})
+
+
+class Riscv2Rs1ImmOperation(Operation):
+    rs1: OpAttr[RegisterAttr]
+    rs2: OpAttr[RegisterAttr]
+    immediate: OpAttr[IntegerAttr]
+
+    @classmethod
+    def get(cls: Type[Op], rs1, rs2, immediate) -> Op:
+        if isinstance(immediate, int):
+            immediate = IntegerAttr.from_int_and_width(immediate, 64)
+        return cls.build(attributes={
+            "rs1": rs1,
+            "rs2": rs2,
+            "immediate": immediate
+        })
+
+
+class Riscv2Rs1OffOperation(Operation):
+    rs1: OpAttr[RegisterAttr]
+    rs2: OpAttr[RegisterAttr]
+    offset: OpAttr[AnyOf([IntegerAttr, LabelAttr])]
+
+    @classmethod
+    def get(cls: Type[Op], rs1, rs2, offset) -> Op:
+        if isinstance(offset, int):
+            offset = IntegerAttr.from_int_and_width(offset, 64)
+        if isinstance(offset, str):
+            offset = LabelAttr.from_str(offset)
+        return cls.build(attributes={"rs1": rs1, "rs2": rs2, "offset": offset})
+
+
+class Riscv1Rd2RsOperation(Operation):
+    rd: OpAttr[RegisterAttr]
+    rs1: OpAttr[RegisterAttr]
+    rs2: OpAttr[RegisterAttr]
+
+    @classmethod
+    def get(cls: Type[Op], rd, rs1, rs2) -> Op:
+        return cls.build(attributes={"rd": rd, "rs1": rs1, "rs2": rs2})
+
+
+class Riscv1Rs1Rt1OffOperation(Operation):
+    rs: OpAttr[RegisterAttr]
+    rt: OpAttr[RegisterAttr]
```

Comment:
What does rt mean? @tobiasgrosser 

Maybe a docstring or two would help in general?